### ISSUE TITLE
order_by title does not work with folder

### ISSUE THREAD

**Issue by ChrisBlankDe** [lukasgeiter/mkdocs-awesome-pages-plugin#73]
the new order_by feature works great with multible files in one directory. Thank you!

But the new feature does not work with directories:

directory tree:
```tree
│   mkdocs.yml
└───docs
    │   .pages
    │   index.md
    ├───1
    │       .pages (Title: A (1))
    │       index.md
    ├───2
    │       .pages (Title: C (2))
    │       index.md
    ├───3
    │       .pages (Title: B (3))
    │       index.md
    └───4
            .pages (Title: D (4))
            index.md
```
.pages at root level:
```yaml
nav: 
  - Home: index.md
  - ... | flat
order_by: title
```

My expection was that the groups where sorted by name (from the .pages in sub dir: A, B, C, D) but they are sorted by the name of the directory (1, 2, 3, 4):
![image](https://user-images.githubusercontent.com/18051656/235715327-4f0ad874-8028-408b-aa68-e692c68da50a.png)

My minimized demo solution:
[mkdocs-awesome-order-by-folder.zip](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/files/11375039/mkdocs-awesome-order-by-folder.zip)

Is this behavior intentional or is it a bug?

**Comment by kamilkrzyskow** [comment]
This is a bug, edge case which I completely overlooked, since I'm not using the `.pages` file to change the title. 
Should probably be an easy fix, because now it's sort of ignoring the custom title from the `.pages` file.

**Comment by kamilkrzyskow** [comment]
Thanks for the reproduction @ChrisBlankDe ✌️ 
![image](https://user-images.githubusercontent.com/34622465/236409924-a2126626-7a7f-4e26-b3cb-0aed0aeb1da0.png)

**Comment by lukasgeiter** [comment]
Fixed in v2.9.1
